Source/WebKit/UIProcess/WebBackForwardCache.cpp

```diff
@@ -48,10 +48,14 @@ WebBackForwardCache::~WebBackForwardCache()
     clear();
 }
 
-inline void WebBackForwardCache::removeOldestEntry()
+void WebBackForwardCache::removeOldestEntry()
 {
-    ASSERT(!m_itemsWithCachedPage.isEmpty());
-    removeEntry(*m_itemsWithCachedPage.first());
+    if (m_itemsWithCachedPage.isEmpty()) {
+        ASSERT_NOT_REACHED();
+        return;
+    }
+    if (auto first = m_itemsWithCachedPage.first())
```

Comment:
Why don't we make `m_itemsWithCachedPage` a WeakHashSet? Seems this new code would still crash if m_itemsWithCachedPage contains a null WeakPtr.

Reply:
They need to be ordered.

Reply:
Alternative in https://github.com/WebKit/WebKit/pull/30057 using WeakListHashSet.